express-notes-api/index.js

```diff
@@ -0,0 +1,68 @@
+const express = require('express');
+const app = express();
+const data = require('./data.json');
+
+app.get('/api/notes', (req, res) => {
+  const notesArr = [];
+  for (const key in data.notes) notesArr.push(data.notes[key]);
+  res.json(notesArr);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  if (req.params.id < 0 || isNaN(req.params.id)) {
+    res.status(400).json({ error: 'id must be a positive number' });
+  } else if (!data.notes[req.params.id]) {
+    res.status(404).json({ error: 'no notes with that id' });
+  } else res.status(200).json(data.notes[req.params.id]);
+});
+
+app.use(express.json());
+
+app.post('/api/notes', (req, res, err) => {
+  if (res.statusCode !== 200) {
+    throw res.status(500).json({ error: 'an unexpected error occured.' });
+  }
+  if (req.body.content === '') {
+    res.status(400).json({ error: 'content is a required field' });
+  } else {
+    const noteObj = req.body;
+    noteObj.id = data.nextId;
+    data.notes[data.nextId++] = noteObj;
+    res.status(201).json(noteObj);
+  }
```

Comment:
After any changes to the `data` you need to save the data back to the `data.json` file so if the server restarts the data will persist.